### PR TITLE
CommentExtensions class for soc771

### PR DESCRIPTION
**Pull request by epishilpa**

### CODE REVIEW

File: src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            var authorReference = String.IsNullOrWhiteSpace(authorId) ?
+                                  Reference.Empty :  
+                                  Reference.Create(String.Format(userReferenceFormat, authorId));
+            var targetReference = Reference.Create(String.Format(resourceReferenceFormat, content.ContentGuid.ToString()));
+
+            var newComment = new Comment(targetReference, authorReference, body, isVisible);
+
+            var service = (new DefaultCommentServiceFactory()).Create();
+            service = ServiceLocator.Current.GetInstance<ICommentService>();
```
Comment (cbanner) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: I don't think that this was intentional -- using the factory, then the service locator, and then finally using the static one? I'm guessing that these two lines (89 & 90) can be removed.

File: src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        private const string resourceReferenceFormat = "resource://{0}";
+        
+        /// <summary>
+        /// Returns the Episerver Social Comment service used for managing comments.
+        /// </summary>
+        private static ICommentService CommentService
+        {
+            get
+            {
+                return ServiceLocator.Current.GetInstance<ICommentService>();
```
Comment (cbanner) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: We shouldn't need to retrieve this each time. We can retrieve a single instance in a static constructor and use that throughout.

Comment (epishilpa) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: @cbanner the property is marked static, do you still prefer I move this to a static ctor?

File: src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs (hunk 1 above)
Comment (epishilpa) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: yes, that was a typo, not intentional. The blog sample looks ok. Thanks for catching this.

File: src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs (hunk 2 above)
Comment (cbanner) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: The property is static but constructs a new comment service with every invocation. I was suggesting that we construct it once and reuse instead.

Comment (epishilpa) [inline src/EPiServer.SocialAlloy.Web/Social/Tutorials/IContent Extensions/CommentExtensions.cs]: @cbanner added static ctor, and updated the blog document too. 
Thanks for reviewing.

**Comment by epishilpa** [comment]
Extending Episerver’s IContent to manage Episerver Social Comments

**Pull request merged**